File: ackindex/ack.pas
function ack (n: byte; x: longint; y: longint): longint;
begin
  if n = 0 then
    ack := y + 1
  else if y = 0 then
    if n = 1 then
       ack := x
    else if n = 2 then
       ack := 0
    else
       ack := 1
  else
     ack := ack(n - 1, x, ack(n, x, y - 1));
end;

begin
  writeln(ack(5, 2, 3));
end.

File: ackindex/ack.js
function ack(n,x,y){
   if(n==0)return y+1;
   if(y==0){
     if(n==1)return x;
     if(n==2)return 0;
     return 1;
   }
   return ack(n-1,x,ack(n,x,y-1));
}
console.log(ack(1,1,9500));
